sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTestUtils.java

```diff
@@ -0,0 +1,136 @@
+ * Utility class for HCatalogIOTest.
+ */
+public class HCatalogIOTestUtils {
+
+  public static final String TEST_TABLE_NAME = "mytable";
+
+  /**
+   * Returns the count of records read from the table.
+   * @param map
+   * @return
+   * @throws HCatException
+   */
+  public static int getRecordsCount(Map<String, String> map) throws HCatException {
+    ReaderContext readCntxt = getReaderContext(map);
+    int recordsCount = 0;
+    for (int i = 0; i < readCntxt.numSplits(); i++) {
+      recordsCount += readRecords(readCntxt, i);
+    }
+    return recordsCount;
+  }
+
+  /**
+   * Returns ReaderContext object for the passed configuration parameters.
+   * @param config
+   * @return
+   * @throws HCatException
+   */
+   public static ReaderContext getReaderContext(Map<String, String> config)
+    throws HCatException {
+    ReadEntity entity = new ReadEntity.Builder().withTable(TEST_TABLE_NAME).build();
+    HCatReader reader = DataTransferFactory.getHCatReader(entity, config);
+    ReaderContext cntxt = reader.prepareRead();
+    return cntxt;
+  }
+
+   /**
+    * Returns WriterContext object for the passed configuration parameters.
+    * @param config
+    * @return
+    * @throws HCatException
+    */
+   public static WriterContext getWriterContext(Map<String, String> config) throws HCatException {
+      WriteEntity.Builder builder = new WriteEntity.Builder();
+      WriteEntity entity = builder.withTable(TEST_TABLE_NAME).build();
+      HCatWriter writer = DataTransferFactory.getHCatWriter(entity, config);
+      WriterContext info = writer.prepareWrite();
+      return info;
+    }
+
+   private static int readRecords(ReaderContext cntxt, int slaveNum) throws  HCatException {
+    HCatReader reader = DataTransferFactory.getHCatReader(cntxt, slaveNum);
+    Iterator<HCatRecord> itr = reader.read();
+    int noOfRecords = 0;
+    while (itr.hasNext()) {
+      ++noOfRecords;
+    }
+    return noOfRecords;
+  }
+
+   /**
+    * Writes records to the table.
+    * @param context
+    * @throws HCatException
+    */
+   public static void writeRecords(WriterContext context) throws HCatException {
+    HCatWriter writer = DataTransferFactory.getHCatWriter(context);
+    writer.write(new HCatRecordItr());
+  }
+
+   /**
+    * Commits the pending writes to the database.
+    * @param config
+    * @param status
+    * @param context
+    * @throws IOException
+    */
+   public static void commitRecords(Map<String, String> config, WriterContext context)
+       throws IOException {
+    WriteEntity.Builder builder = new WriteEntity.Builder();
+    WriteEntity entity = builder.withTable(TEST_TABLE_NAME).build();
+    HCatWriter writer = DataTransferFactory.getHCatWriter(entity, config);
+    writer.commit(context);
+  }
+
+   /**
+    * Returns DefaultHCatRecord for passed value.
+    * @param value
+    * @return
+    */
+   public static DefaultHCatRecord toHCatRecord(int value) {
+    List<Object> dataList = new ArrayList<Object>(2);
```

Comment:
Can be condensed into one line: `return new DefaultHCatRecord(Arrays.asList("record " + value), value)`